### ISSUE TITLE
False-positive test fail

### ISSUE THREAD

**Issue by EvgenyOrekhov** [tapjs/tapjs#470]
I have two objects that are strictly same, but tap says they are not.

Demo: https://repl.it/repls/StickyThoughtfulCodegeneration

If I stringify the objects, there is no error, they are the same:

```js
        t.strictSame(
            JSON.stringify(result),
            JSON.stringify(expected),
            "should lint using JSLint"
        );
```

But later when I compare non-stringified objects, tap says

```text
  not ok should lint using JSLint
    +++ found                                                           
    --- wanted                                                          
  
    at:
        line: 36
        column: 11
        file: index.js
      stack: |
        index.js:36:11
      source: |
        t.strictSame(
```

What is going on?

**Comment by ovhemert** [comment]
The documentation of `t.strictSame()` says:

_Verify that the found object is deeply equivalent to the wanted object._
_Use strict equality for scalars (ie, ===)._

So if you do a compare like `result === expected`, you would also get `false`.
This is, because you are comparing objects created separately. They may have the same properties, but are created at separately, and thus have a different reference, which makes them different in a compare with `===`.

Example:
```
let a = { x: 1 }
let b = { x: 1 }
a===b // => false
```
 
Converting the objects to a string with `JSON.stringify()` creates 2 strings that are identically the same. In the case of `===` with 2 string values, they are the same.

So, if you want to compare 2 different objects with the same properties, stick with the `JSON.stringify()` solution if that works for you.

**Comment by EvgenyOrekhov** [comment]
@ovhemert I can not reproduce your example, tap says

> ok 1 - should be equivalent strictly

https://repl.it/repls/UnimportantThornyKilobyte

You are describing how [t.equal()](https://www.node-tap.org/asserts/#tequalfound-wanted-message-extra) works. t.strictSame() does not compare objects with `===`.

**Comment by isaacs** [comment]
The issue here is that the objects are being created in a different context, so the Array constructor didn't match.  Confusing to debug, because they both JSON the same, as you noted!

Bumping to tsame 2.0.1 fixes the problem.  I'll drop a new tap release shortly with the fix.

**Comment by isaacs** [comment]
Fixed in tap 12.0.2.
